addons/website/models/res_users.py

```diff
@@ -72,10 +72,14 @@ def authenticate(cls, db, login, password, user_agent_env):
                 visitor_sudo = env['website.visitor']._get_visitor_from_request()
                 if visitor_sudo:
                     partner = env.user.partner_id
-                    partner_visitor = env['website.visitor'].sudo().search([('partner_id', '=', partner.id)])
+                    partner_visitor = env['website.visitor'].with_context(active_test=False).sudo().search([('partner_id', '=', partner.id)])
                     if partner_visitor and partner_visitor.id != visitor_sudo.id:
+                        # Link history to older Visitor and delete the newest
                         visitor_sudo.website_track_ids.write({'visitor_id': partner_visitor.id})
                         visitor_sudo.unlink()
+                        # Reactivate the older Visitor & Update visitor's Cookie (TODO: check if necessary (maybe done in _handle_webpage_dispatch)
```

Comment:
1. I don't like that TODO, can you actually check?
2. It's unclear here that visitors are archived by a cron, should be explained in the docstring AND in the commit message
3. I don't see any cookie update, the comment is a little confusing
4. I attach an image related to all points above

![brain](https://user-images.githubusercontent.com/41282536/69150453-fa7e0d00-0ad8-11ea-958f-812adf2c6a8a.png)